Rakefile

```diff
@@ -86,3 +86,211 @@ EOM
+      },
+      'vars' => {
+        'is_servicenow_instance' => true
+      }
+    }]
+    write_to_inventory_file(inventory_hash, 'inventory.yaml')
+  end
+
+  desc 'Sets up the ServiceNow CMDB with entries for each VM'
+  task :setup_servicenow_cmdb do
+    if sys_id = cmdb_record_ids[on_master(&:uri)]
+      # Double-check that the CMDB record exists on the ServiceNow instance
+      task_result = on_servicenow_instance do |instance|
+        instance.run_bolt_task(
+          'servicenow_tasks::get_record',
+          { 'table' => 'cmdb_ci', 'sys_id' => sys_id },
+          expect_failures: true,
+        )
+      end
+      if error = task_result.result['error']
+        raise "inventory.yaml reports a sys_id of #{sys_id} for master's CMDB record, but servicenow_tasks::get_record fails to retrieve the record: #{error}"
+      end
+      puts("A CMDB record's already been created for the master (sys_id = #{sys_id})")
+      next
+    end
+
+    # CMDB record doesn't exist so create it
+    puts("Creating the master's CMDB record ...")
+    cmdb_record = JSON.parse(File.read('spec/support/acceptance/cmdb_record_template.json'))
+    cmdb_record['fqdn'] = on_master(&:uri)
+    task_result = on_servicenow_instance do |instance|
+      instance.run_bolt_task(
+        'servicenow_tasks::create_record',
+        { 'table' => 'cmdb_ci', 'fields' => cmdb_record } 
+      )
+    end
+    sys_id = task_result.result['result']['sys_id']
+
+    # Add the CMDB record to the inventory.yaml file
+    master_uri = on_master(&:uri)
+    inventory_hash = inventory_hash_from_inventory_file
+    inventory_hash['vars'] = {
+      'cmdb_record_ids' => cmdb_record_ids.merge(master_uri => sys_id),
+    }
+    puts("Adding the created CMDB record (sys_id = #{sys_id}) to inventory['vars'] ...")
+    write_to_inventory_file(inventory_hash, 'inventory.yaml')
+  end
+
+  desc 'Installs the module on the master'
+  task :install_module do
+    on_master { |master| Rake::Task['litmus:install_module'].invoke(master.uri) }
+  end
+
+  desc 'Set up the test infrastructure'
+  task :setup do
+    tasks = [
+      :provision_vms,
+      :setup_pe,
+      :setup_servicenow_instance,
+      :setup_servicenow_cmdb,
+      :install_module,
+    ]
+
+    tasks.each do |task|
+      task = "acceptance:#{task}"
+      puts("Invoking #{task}")
+      Rake::Task[task].invoke
+      puts("")
+    end
+  end
+
+  desc 'Runs the tests'
+  task :run_tests do
+    puts("Running the tests ...\n")
+
+    # TODO: We can't use litmus:acceptance:parallel here b/c that will try
+    # to run the spec tests on _all_ targets in the inventory.yaml file,
+    # including our ServiceNow instance. Until litmus:acceptance:parallel
+    # has a way for you to pass-in specific targets to run the tests on,
+    # we'll just run these tests via rspec
+    system({ 'TARGET_HOST' => on_master(&:uri) }, 'bundle exec rspec ./spec/acceptance')
+  end
+
+  desc 'Teardown the setup'
+  task :tear_down do
```

Comment:
Is this really necessary? The mock server and the master are both running on the same pooler vm aren't they? Just call litmus' tear down task and it will throw away the vm. Why go to all this trouble to clean up a vm that's about to thrown away?

Reply:
This also clears out test-CMDB records if I end up using something other than the mock ServiceNow instance (like my dev instance). Notice that that code noops if the user ends up using the mock ServiceNow instance. So, I would like to keep it.